### ISSUE TITLE
Misleading document started using Tensorflow terminology in the document to describe the Pytorch Transformer code.

### ISSUE THREAD

**Issue by JonathanSum** [pytorch/pytorch#82041]
### 📚 The doc issue

![image](https://user-images.githubusercontent.com/21982975/180585585-6077a456-c2d9-4e15-a1f6-014b93feb13b.png)
 the model must be executed in inference mode and operate on input tensors that do not collect gradient tape information (e.g., running with torch.no_grad).

### Suggest a potential alternative/fix

 the model must be executed in inference mode and operate on input tensors that do not collect gradient tape information (e.g., running with torch.no_grad).
Change it to be:
the model must be executed in inference mode and operate on input tensors that does not accumulate gradient. (e.g, setting the model with torch.no_grad).
